src/plugins/plot/axis/XAxis.vue

```diff
@@ -137,15 +137,17 @@ export default {
             const xAxisKey = this.xAxis.get('key');
 
             this.xKeyOptions = this.seriesModel.metadata
-                .valuesForHints(['domain'])
-                .map(function (o) {
-                    return {
-                        name: o.name,
-                        key: o.key
-                    };
-                });
+                ? this.seriesModel.metadata
```

Comment:
Since this is such a big block of code, can we set this.xKeyOptions to a `[]` right off the bet, then throw the actual assignment in an if block, I think that will make it more readable than a tertiary.